### PR TITLE
Add better error reporting when layers are in use

### PR DESCRIPTION
**Pull request by rhatdan**
We need to inform user what object is using a layer when returning and
error.

Signed-off-by: Dan Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
This patch is to help diagnose issues like https://github.com/projectatomic/buildah/issues/134

File: store.go [hunk 1]
```diff
@@ -1255,8 +1255,8 @@ func (s *store) DeleteImage(id string, commit bool) (layers []string, err error)
 		for _, container := range containers {
 			aContainerByImage[container.ImageID] = container.ID
 		}
-		if _, ok := aContainerByImage[id]; ok {
-			return nil, ErrImageUsedByContainer
+		if container, ok := aContainerByImage[id]; ok {
+			return nil, errors.Wrap(ErrImageUsedByContainer, "Image used by %v", container.ID)
```
Comment (TomSweeneyRedHat) [inline store.go]: Should this be WrapF?

Comment (rhatdan) [inline store.go]: yes

File: store.go [hunk 2]
```diff
@@ -1916,7 +1916,7 @@ func (s *store) Shutdown(force bool) ([]string, error) {
 		}
 	}
 	if len(mounted) > 0 && err == nil {
-		err = ErrLayerUsedByContainer
+		err = errors.Wrapf(ErrLayerUsedByContainer, "A layer is mounted")
```
Comment (nalind) [inline store.go]: This can be `errors.Wrap()`.

**Comment by rhatdan** [comment]
@nalind Fixed to use Wrap instead of Wrapf.

**Comment by nalind** [comment]
LGTM, merging.

**Pull request merged**